package.json

```diff
@@ -26,6 +26,7 @@
   },
   "dependencies": {
     "@babel/plugin-proposal-class-properties": "^7.12.1",
+    "@material-ui/core": "^4.11.2",
```

Comment:
Wait so we _are_ adding `material-ui` anyway? After the whole back and forth about whether to use it or not 😂

Reply:
Was going to use it for the popover, but @tgolen convinced us not to. Not sure how to advise in this case. I think we decided that once the `material-ui` doors were open then people would feel comfortable using all sorts of things from it and potentially making a mess (like `jQuery-UI`) or only using one thing from it (like our `DatePicker`). Also increases to the bundle size.

Reply:
> Wait so we are adding material-ui anyway? After the whole back and forth about whether to use it or not 😂

Yeah, I thought last we left off we decided we were going to use it because it was the simplest solution to solve the problems we have presently. I _do_ have a backup plan but it's going to be complex and janky by comparison. IDK much about using Popper.js but I doubt that would be much better than using `material-ui`.

I think we could keep the scope of things using `material-ui` pretty small in that it can only be used for features which will only ever be web-only. I was going to use it for the popover-style ReportActionContextMenu too

Reply:
> I thought last we left off we decided we were going to use it 

Where was this decided? I am still extremely opposed to doing this.

Matariel UI is **_not_** a good route to take:
1. It's an extremely large library (86k which is more than 10% of our entire application)
2. It does hundreds of things we will never use it for
3. It conflicts with all our past history of standardizing on bootstrap
4. Lessons were learned from the importing (and now impossible to maintain) jQuery-UI import that this path should be avoided at all costs
5. It's not cross-platform

> IDK much about using Popper.js but I doubt that would be much better than using material-ui.

There are several benefits to using pooper.js (ok, popper.js):
1. It's an extremely small library a fraction of the size of material UI
2. It only does one thing, popovers, and that's all we need (material does hundreds of things that will never be used)

**But I don't want to use pooper.js either**

# Suggested Solution
Build our own popovers.

1. They are not difficult to build. All they are is a modal with slightly different positioning, styling, and triggers
2. We should always consider building our own components that meet our exact needs rather than tethering ourselves to third-party alternatives. Yes, I know there are examples of us doing the opposite, but removing this is our first instinct will allow the app to maintain a small size, keep it as maintainable as the rest of the code, and ensure that solutions are cross-platform.

Reply:
`pooper.js` 😂  

I'll take some heat for that decision since it was me who convinced everyone in the doc that using a JS library would be better than creating our own positioning library. @roryabraham your initial instinct was to take @tgolen's **Suggested Solution** here and I'm sorry for steering you off course. How close were we to the home brewed solution before that happened?

Reply:
Maybe some more context about what is "complex and janky" would be good? Happy to help work with ya on it.

Reply:
As you all know by now, we've got a home-baked tooltip solution in the works.